packages/modal/src/Modal.js

```diff
@@ -0,0 +1,225 @@
+  }
+
+  &:focus {
+    outline: none;
+  }
+`
+
+const FloatCloseButton = styled(CloseButton)`
+  position: absolute;
+  top: ${props => (props.header ? '24px' : '16px')};
+  right: 16px;
+  z-index: 2;
+
+  svg {
+    vertical-align: top;
+  }
+
+  &:focus {
+    outline: none;
+  }
+`
+
+const ContentWrapper = styled(Box)`
+  position: relative;
+  ${props => {
+    if (!props.enableoverflow) {
+      if (props.header === 1) {
+        return `
+          overflow: scroll;
+          height: calc(100% - 8px);
+        `
+      } else if (props.header === 2) {
+        return `
+          overflow: scroll;
+          height: calc(100% - 40px);
+        `
+      } else {
+        return `
+          overflow: scroll;
+          height: 100%;
+        `
+      }
+    }
+  }}
+`
+
+const OverlayWrapper = styled.div`
+  display: table;
+  height: 100%;
+  width: 100%;
+`
+
+const DialogWrapper = styled.div`
+  display: table-cell;
+  vertical-align: middle;
+  ${props =>
+    props.enableoverflow &&
+    `
+    padding-top: 24px;
+    padding-bottom: 24px;
+  `};
+`
+
+const Modal = ({
+  isOpen,
+  onClose,
+  bg,
+  zIndex,
+  children,
+  title,
+  headerBg,
+  imgMode,
+  width,
+  className,
+  header,
+  disableCloseButton,
+  enableOverflow
+}) => (
+  <Transition
+    items={isOpen}
+    config={config.gentle}
+    from={{ opacity: 0, transform: 'scale(0.5)' }}
+    enter={{ opacity: 1, transform: 'scale(1)' }}
+    leave={{ opacity: 0, transform: 'scale(0.5)' }}
+  >
+    {isOpen =>
+      isOpen &&
+      (styles => (
+        <Overlay
+          onDismiss={onClose}
+          zindex={zIndex}
+          style={{ opacity: styles.opacity }}
+        >
+          <OverlayWrapper>
+            <DialogWrapper enableoverflow={enableOverflow}>
+              <Dialog
+                width={width}
+                bg={bg || 'white'}
+                style={{ transform: styles.transform }}
+                className={className}
+                enableOverflow={enableOverflow}
+              >
+                {header && (
+                  <StyledHeader
+                    id="pcln-modal-header"
+                    header={header}
+                    align="center"
+                    color="white"
+                    bg={headerBg || 'blue'}
+                    pl="16px"
+                  >
+                    {header === 2 && title && (
+                      <Text fontSize={1} bold>
+                        {title}
+                      </Text>
+                    )}
+                    {!disableCloseButton && header === 2 && (
+                      <StyledCloseButton
+                        id="pcln-modal-close"
+                        onClick={onClose}
+                        ml="auto"
+                      />
+                    )}
+                  </StyledHeader>
+                )}
+                {header !== 2 && !disableCloseButton && (
+                  <FloatCloseButton
+                    id="pcln-modal-close"
```

Comment:
Is there a reason you need this to be an ID? 
We should avoid using ids to make these components as resuable as possible.

Reply:
@rykimchi the ids are not used by the modal component itself at all. the purpose of ids is that the modal has so many layers, simply accepting className makes style change to deeper level hard, so these ids are helping people on that. Open to other suggestion.